src/driver/swapchain.rs

```diff
@@ -181,7 +181,7 @@ impl Swapchain {
 
     /// Gets information about this swapchain.
     pub fn info(&self) -> SwapchainInfo {
-        self.info
+        self.info.clone()
```

Comment:
Not sure if clone is preferable here or returning a reference. Returning a reference would change the API surface, but could be preferable.

Reply:
I'm going to change all of these soon. The info is available in the other types as a public field, but that's not great because you may make changes to info after the thing has been created. I'd like to use @dtolnay's public field deref method to make them available but read-only.